lib/plugins/aws/deploy/lib/uploadCloudFormationTemplate.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  uploadCloudFormationTemplateToS3Bucket() {
+    this.serverless.cli.log('Uploading CloudFormation file to S3...');
+
+    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
+
+    const params = {
+      Bucket: this.bucketName,
+      Key: 'compiled-cloudformation-template.json',
+      Body: body,
+    };
+
+    return this.sdk.request('S3',
+      'putObject',
+      params,
+      this.options.stage,
+      this.options.region);
+  },
+
+  uploadCloudFormationTemplate() {
```

Comment:
Yes. You're right. This was some code I wrote because I first thought about keeping 5 versions of the CloudFormation template and adding steps like `cleanup` etc. but then moved on to store just one CloudFormation template in the bucket.